zerver/openapi/test_curl_examples.py

```diff
@@ -49,22 +49,15 @@ def test_generated_curl_examples_for_success(client: Client, owner_client: Clien
                 curl_command_html = md_engine.convert(line.strip())
                 unescaped_html = html.unescape(curl_command_html)
                 curl_command_text = unescaped_html[len("<p><code>curl\n") : -len("</code></p>")]
+                # TODO: To delete another user we change the authentication_line.
+                change_authentication_line = False
+                if file_name == "templates/zerver/api/deactivate-own-user.md":
+                    change_authentication_line = True
+                    authentication_line = f"{delete_test_client.email}:{delete_test_client.api_key}"
```

Comment:
While this does seem less hacky, the point of the refactoring is also to remove endpoint-specific stuff(like `needs_reactivation` and `change_authentication_line`) from the general test code(I'd suggest you look into `curl_param_value_generators.py` and see if we can make some changes there for this to work).

Reply:
Oh, I got your point. I'll update PR.